### DISCUSSION TITLE
Reads package org.mapsforge.map.rendertheme from both mapsforge.map and mapsforge.themes

### DISCUSSION THREAD

**Discussion by tobithe1st** [mapsforge/mapsforge#1414] [Q&A] [answered]
Hey guys,
I am just getting started with mapsforge, but when I try to run my SampleApplication the following error occurs:
`module com.henrytriage reads package org.mapsforge.map.rendertheme from both mapsforge.map and mapsforge.themes`

That's the mapsforge part of my maven pom.xml:
`

        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-core</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-map</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-map-reader</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-themes</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-map-awt</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>guru.nidi.com.kitfox</groupId>
            <artifactId>svgSalamander</artifactId>
            <version>1.1.3</version>
        </dependency>
        <dependency>
            <groupId>net.sf.kxml</groupId>
            <artifactId>kxml2</artifactId>
            <version>2.3.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-poi</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.mapsforge</groupId>
            <artifactId>mapsforge-poi-awt</artifactId>
            <version>0.19.0</version>
        </dependency>
        <dependency>
            <groupId>org.xerial</groupId>
            <artifactId>sqlite-jdbc</artifactId>
            <version>3.28.0</version>
        </dependency>

`

and my module-info.java:
`    requires javafx.controls;
    requires javafx.fxml;
            
    requires org.controlsfx.controls;
    requires com.dlsc.formsfx;
    requires net.synedra.validatorfx;
    requires org.kordamp.ikonli.javafx;
    requires org.kordamp.bootstrapfx.core;
    requires java.sql;
    requires java.desktop;
    requires javafx.web;
    requires javafx.swing;
    requires org.apache.commons.io;
    requires mapsforge.core;
    requires mapsforge.map.awt;
    requires mapsforge.map;
    requires mapsforge.map.reader;
    requires java.prefs;
    requires mapsforge.themes;`

Thanks in advance for any help!

**Comment by devemux86** [comment]
Have not seen such problem on desktop, even with Gradle. You could try:
- Declare only the desktop dependency, it should find its children
e.g. `mapsforge-map-awt`
- Or declare all dependencies and disable the transitive for Mapsforge
```xml
<exclusions>
    <exclusion>
         <groupId>*</groupId>
        <artifactId>*</artifactId>
    </exclusion>
</exclusions>
```

**Reply by tobithe1st** [reply]
I need the InternalRenderTheme, therefore I import `org.mapsforge.map.rendertheme.InternalRenderTheme;`
... so the first suggestion doesn't work
also the second solution didn't work

I think the problem is, that the package of `rendertheme `in `org.mapsforge:mapsforge-map` is called `org.mapsforge.map.rendertheme`
and there also is a package named the same in `org.mapsforge:mapsforge-themes`

@devemux86 definitely thx for you help!

**Reply by devemux86** [reply, marked as answer]
Probably now Java does not like 2 modules to have similar packages? Which would be very strange!

There should probably exist an option in the Java build environments to disable such strict limitation.

**Reply by tobithe1st** [reply]
I now solved it by implementing the mapsforge-themes locally into my project...
